components/BottomNavigation/examples/MDCBottomNavigationBarController.m

```diff
@@ -0,0 +1,265 @@
+    return [_delegate bottomNavigationBarController:self
+                         shouldSelectViewController:viewControllerToSelect];
+  }
+
+  return YES;
+}
+
+#pragma mark - Private Methods
+
+/**
+ * Removes the given view controller from its parent view controller and its view from its superview
+ */
+- (void)removeContentViewController:(UIViewController *)viewController {
+  [viewController removeFromParentViewController];
+  [viewController.view removeFromSuperview];
+}
+
+/**
+ * Adds the given view controller to the view controller hierarchy and its view to the content
+ * view.
+ */
+- (void)addContentViewController:(UIViewController *)viewController {
+  BOOL doesNotContainViewController = ![self.childViewControllers containsObject:viewController];
+  if (viewController && doesNotContainViewController) {
+    [self addChildViewController:viewController];
+    [self.content addSubview:viewController.view];
+    [self addConstraintsForContentView:viewController.view];
+    [viewController didMoveToParentViewController:self];
+  }
+}
+
+/**
+ * Deselects the currently set item.  Sets the selectedIndex to NSNotFound, the naviagation bar's
+ * selected item to nil, and the selectedViewController to nil.
+ */
+- (void)deselectCurrentItem {
+  _selectedIndex = NSNotFound;
+  _navigationBar.selectedItem = nil;
+
+  // Force removal of the currently selected viewcontroller if there is one.
+  self.selectedViewController = nil;
+}
+
+/**
+ * Sets the selected view controller to the corresponding index and updates the navigation bar's
+ * selected item.
+ */
+- (void)updateViewsForSelectedIndex:(NSUInteger)index {
+  // Update the selected view controller
+  UIViewController *selectedViewController = [_viewControllers objectAtIndex:index];
+  self.selectedViewController = selectedViewController;
+
+  // Update the navigation bar's selected item.
+  _navigationBar.selectedItem = selectedViewController.tabBarItem;
+  [self setNeedsStatusBarAppearanceUpdate];
+}
+
+/**
+ * Hooks up the constraints for the subviews of this controller.  Namely the content view and the
+ * navigation bar.
+ */
+- (void)loadConstraints {
+  _content.translatesAutoresizingMaskIntoConstraints = NO;
+  _navigationBar.translatesAutoresizingMaskIntoConstraints = NO;
+
+  // Navigation Bar Constraints
+  [self.view.leftAnchor constraintEqualToAnchor:_navigationBar.leftAnchor].active = YES;
+  [self.view.rightAnchor constraintEqualToAnchor:_navigationBar.rightAnchor].active = YES;
+
+  _navigationBarHeightConstraint =
+      [_navigationBar.heightAnchor constraintEqualToConstant:[self calculateNavigationBarHeight]];
+  _navigationBarHeightConstraint.active = YES;
+
+  [_navigationBar.topAnchor constraintEqualToAnchor:_content.bottomAnchor].active = YES;
+  [_navigationBar.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor].active = YES;
+
+  [_navigationBar setContentHuggingPriority:UILayoutPriorityDefaultHigh
```

Comment:
Is this required given the left/right constraints?

Reply:
Correct.  Removed.